internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -321,15 +322,22 @@ func (connector *SingleUseConnector) writeErrorToClient(err error) {
 }
 
 func dataSourceName(connDetails *ConnectionDetails) string {
+	dsn := "sqlserver://%s:%s@%s?%s"
+
+	var params []string
+	for key, value := range connDetails.SSLParams {
+		params = append(params, fmt.Sprintf("%s=%s", key, value))
+	}
+
 	return fmt.Sprintf(
 		// NOTE: as things stand the resulting DSN here means that if TLS is available
 		// it'll be used between Secretless and the server.
 		// To disable TLS altogether we must change to
 		// "sqlserver://%s:%s@%s?encrypt=disable",
-		"sqlserver://%s:%s@%s?encrypt=%s",
+		dsn,
 		connDetails.Username,
 		connDetails.Password,
 		connDetails.Address(),
-		connDetails.SSLMode,
+		strings.Join(params, "&"),
```

Comment:
I think we should refactor the code here to something like below. By using the url package to generate the url string we get encoding for free. This is mostly for the `rawcertificate` which will get damaged in transit if we don't encode.
```
query := url.Values{}
var params []string
for key, value := range connDetails.SSLParams {
	query.Add(key, value)
}

u := &url.URL{
	Scheme:   "sqlserver",
	User:     url.UserPassword(connDetails.Username, connDetails.Password),
	Host:     connDetails.Address(),
	RawQuery: query.Encode(),
}

return u.String()
```